Source/IGListAdapterDataSource.h

```diff
@@ -60,7 +60,27 @@ NS_SWIFT_NAME(ListAdapterDataSource)
  adding the background view and maintaining its visibility.
  */
 - (nullable UIView *)emptyViewForListAdapter:(IGListAdapter *)listAdapter;
-
+    
+@optional
```

Comment:
just noting:

we can keep this for `3.2` to avoid breaking clients and remove for `4.0`.

after this merges, let's be sure to open a new issue to track this.